Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_DI.cpp

```diff
@@ -19,13 +20,81 @@
 
 using namespace DVDInterface;
 
-CWII_IPC_HLE_Device_di::CWII_IPC_HLE_Device_di(u32 _DeviceID, const std::string& _rDeviceName )
+static int ioctl_callback;
+static std::deque<u32> m_commands_to_execute;
+
+static void InternalIOCtl(u32 command_address)
+{
+	u32 BufferIn = Memory::Read_U32(command_address + 0x10);
+	u32 BufferInSize = Memory::Read_U32(command_address + 0x14);
+	u32 BufferOut = Memory::Read_U32(command_address + 0x18);
+	u32 BufferOutSize = Memory::Read_U32(command_address + 0x1C);
+
+	u32 command_0 = Memory::Read_U32(BufferIn);
+	u32 command_1 = Memory::Read_U32(BufferIn + 4);
+	u32 command_2 = Memory::Read_U32(BufferIn + 8);
+
+	DEBUG_LOG(WII_IPC_DVD, "IOCtl Command(0x%08x) BufferIn(0x%08x, 0x%x) BufferOut(0x%08x, 0x%x)",
+	          command_0, BufferIn, BufferInSize, BufferOut, BufferOutSize);
+
+	// TATSUNOKO VS CAPCOM: Gets here with BufferOut == 0!!!
+	if (BufferOut != 0)
+	{
+		// Set out buffer to zeroes as a safety precaution
+		// to avoid answering nonsense values
+		Memory::Memset(BufferOut, 0, BufferOutSize);
+	}
+
+	// The original hardware overwrites the command type with the async reply type.
+	Memory::Write_U32(IPC_REP_ASYNC, command_address);
+	// IOS also seems to write back the command that was responded to in the FD field.
+	Memory::Write_U32(Memory::Read_U32(command_address), command_address + 8);
```

Comment:
Might be a little more clear to move these writes to IOCtlCallback.  (IIRC, it doesn't actually matter when they run, so might as well do all the writing at the same time.)